src/main.go

```diff
@@ -248,3 +245,79 @@ func newClaimsValidationFn(allowedTenantIDs []string) options.ClaimsValidationFn
 		return nil
 	}
 }
+
+type cognitoClaims struct {
+	Subject   string    `json:"sub"`
+	TokenUse  string    `json:"token_use"`
+	Scope     string    `json:"scope"`
+	AuthTime  time.Time `json:"auth_time"`
+	Issuer    string    `json:"iss"`
+	ExpiresAt time.Time `json:"exp"`
+	IssuedAt  time.Time `json:"iat"`
+	Version   int       `json:"version"`
+	Jti       string    `json:"jti"`
+	ClientID  string    `json:"client_id"`
+}
+
+func newCognitoClaimsValidationFn(requiredScope string) options.ClaimsValidationFn[cognitoClaims] {
+	return func(c *cognitoClaims) error {
+		if c.TokenUse != "access" {
+			return fmt.Errorf("expected token_use to be access but received: %s", c.TokenUse)
+		}
+
+		if c.Scope != requiredScope {
+			return fmt.Errorf("expected scope to be %s but received: %s", requiredScope, c.Scope)
+		}
+
+		return nil
+	}
+}
+
+type tokenValidator func(ctx context.Context, tokenString string) error
+
+func newTokenValidator(cfg config) (tokenValidator, error) {
+	azureTenantValidator, err := oidctoken.New(
+		cfg.azureValidationFn,
+		options.WithRequiredTokenType(cfg.azureTokenType),
+		options.WithIssuer(cfg.azureIssuer),
+		options.WithRequiredAudience(cfg.azureAudience),
+		options.WithDisableIssuerValidation(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	cognitoValidators := []*oidctoken.TokenHandler[cognitoClaims]{}
+	for _, issuer := range cfg.AllowedCognitoIssuers {
+		cognitoValidator, err := oidctoken.New(
+			newCognitoClaimsValidationFn(cfg.cognitoRequiredScope),
+			options.WithIssuer(issuer),
+		)
+		if err != nil {
+			return nil, err
+		}
+		cognitoValidators = append(cognitoValidators, cognitoValidator)
+	}
+
+	validatorFn := func(ctx context.Context, tokenString string) error {
```

Comment:
I'm a bit confused here. Don't you want a lookup table on issuer, so you can ask the right validator? This way, a failed validation will have 100 "not my monkey" errors?